### DISCUSSION TITLE
Default repository in GitHub Project?

### DISCUSSION THREAD

**Discussion by Gary-H9** [community/community#60258] [Projects and Issues]
### Select Topic Area

Question

### Feature Area

Projects

### Body

👋  Hello, 

I'd like to be able to create Issues in a _default_ repo when I'm using GitHub Projects, I'm not sure if it is possible. Is it? 

Specifically if I click `+ Add Item` I always have to select my repository but with a default option I wouldn't need to do this. 

Ideally, I think it would be good to be able to set a default repository for a project in the Settings page for a Project. 
`https://github.com/users/<username>/projects/<project_number>/settings`

If not is this something that can be done now, could it be considered as a feature request? It's a very minor thing but I think it would improve the user experience for people who are using a GitHub Project to manage work in one repository.

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by sohaibaumer** [comment]
Currently, GitHub Projects does not have a default repository setting for creating issues. You need to select the repository each time when adding an item to a project board. However, you can consider submitting a feature request to GitHub Support or the GitHub Community forums to suggest adding this feature. It can enhance the user experience for managing work in a single repository.

**Reply by queenofcorgis** [reply]
Hi @Gary-H9 - I've tagged this as product feedback so our team can review, thank you for submitting this request 🚀

**Reply by usirg** [reply]
any hope for this feature? there is definitely [demand](https://github.com/orgs/community/discussions/8251)

**Reply by oddballpete** [reply]
Following up here; the Department of Veteran Affairs (US Govt) is in the process of adopting Github projects, and this is posing an issue. We would like to define the project an issue belongs to on creation. Could we request an update / escalation for this please.

**Comment by dallenbaldwin** [comment]
Having a default Repository setting would be super nice for smaller, simpler teams. 

When a default repository is set, I would love to see quick links to different pages in the repository like Pull Requests, Code, Actions, etc.

**Comment by Th3S4mur41** [comment]
Weirdly enough it seems to currently use the earliest repository (creation date) as default.
In one of my orgs, this happens to be an archived repository where issue creation doesn't even work.
When you click on the create issue button, nothing happens.

So, the current state is not only inconvenient, but also buggy.

**Comment by pzgz** [comment]
@Th3S4mur41 is correct, the problem is not only lack of ability to set default repo for a project, it do have a default repo selected whenever creating an issue, even the repo was already archived. It means, everytime I create an issue, have to popup the repo dropdown list and then select the only one which is not archieved. In my case, I have three projects in my organization and only one is active, but it just default to the one which shouldn't be there.

**Comment by luca-domenichini** [comment]
This is super annoying!

Also adding that if you archive the "earliest" repo, create issue dialog still wants to create issues on that repo.

Deleting the "earliest" repo is the current, well..., workaround.

**Comment by kfidrin** [comment]
@queenofcorgis any updates? This issue is super annoying

**Comment by jnankin** [comment]
can we get this in please

**Comment by imsgx** [comment]
+1

**Comment by astanciu** [comment]
The current way to enter issues just has too much friction, too many clicks. I don't need to select a repo and template every time.

**Comment by karimelmel** [comment]
In itself a reason to not use GitHub Projects

**Comment by CollinTerSteegeVBP** [comment]
Please add this feature, it would make our lives much easier

**Comment by delai** [comment]
We’ve also been troubled by this issue for a long time, but recently we finally found a simple workaround:

Rename your default repo (for example, from A to B), then create a new repo called A. After that, in your local code repository (which was originally pointing to A), run:

```
git push -u origin main
```

Hope this helps.

**Comment by Shubham-cyber-prog** [comment]
Currently ,  GitHub Projects (beta or classic ) don't support setting a "default repository " directly . Projects can include issues or PRs from multiple repositories which makes flexible across orgs or teams 

if your projects mostly tracks issues from a single repository , use filters like ; 

Features Request Idea : It would be really helpful if GitHub allowed setting a "default repo" for a project so new issues could automatically link to it , especially for mono-repo teams 

Unitil then , saved filters and views can help simulate this behavior 
Hope this helps others

**Reply by historictors** [reply]
thanks for information

**Reply by Rajsingh89** [reply]
good information brother

**Reply by vanshrajv** [reply]
good information brother

**Comment by anik8das** [comment]
+1! this is much needed. Is making my life miserable in an early stage company with projects across different repos
Filters do simplify the view but default repository is the main point of friction

**Comment by bfops** [comment]
GitHub recently rolled out changes to bypass the repo selection screen entirely. While this addresses the complaints of people who wanted to minimize clicks, it has created a new pain point - namely, that the GitHub-selected unchangeable default repo is often not the right one.

This is particularly painful when the default repo is public, but there are private repos in the project. In the short time since this change, I have seen this lead to numerous instances where public issues have been created with sensitive content.

GitHub project management honestly continues to disappoint. It's buggy, slow, and randomly missing important functionality.

**Comment by ruffsl** [comment]
Also, if no default repo is configured for the project, it would be handy if the repo dropdown at least remembered your most recent selection(s), so when frequently creating multiple tickets, each contributor could at least benefit from some personal UX caching.

There's already user specific tracking of recently viewed projects, so adding tracking for recently selected repos mapped for each project doesn't seem much of a stretch from a UX pattern. I'd just like the actively selected repo from the initial state of the dropdown during ticket creation to be the most recently selected option - per user, or overridden to be the default repo via project's setting.

**Reply by pzgz** [reply]
IMHO, remember most recent selection might be way more difficult compare to default repo or ignore archived repo, just getting this judgement based on their attitude... I don't think they care at all.

**Comment by luca-domenichini** [comment]
as of 2025-11-10, I see this:
<img width="527" height="591" alt="image" src="https://github.com/user-attachments/assets/db3103f3-bee7-427f-92cd-c53ef7e2c28e" />

Anyone knows when was this added? This perfectly solves my problems!

**Reply by ghost** [reply]
Very nice. 

Now all I would like is to be able to do this via the GitHub CLI
